src/features/journeys/store.ts

```diff
@@ -44,6 +44,7 @@ const journeysSlice = createSlice({
       const instanceItem = state.journeyInstanceList.items.find(
         (item) => item.id == instanceId
       );
+      state.milestonesByInstanceId[instanceId].isStale = true;
```

Comment:
`state.milestonesByInstanceId[instanceId]` can be `undefined`, if the user has not visited the milestones page before interacting with the journey some other way.